automerge/src/transaction/inner.rs

```diff
@@ -405,15 +422,48 @@ impl TransactionInner {
         let obj = doc.exid_to_obj(ex_obj)?;
         for _ in 0..del {
             // del()
-            self.local_op(doc, obj, pos.into(), OpType::Delete)?;
+            self.local_op(doc, op_observer, obj, pos.into(), OpType::Delete)?;
         }
         for v in vals {
             // insert()
-            self.do_insert(doc, obj, pos, v.clone().into())?;
+            self.do_insert(doc, op_observer, obj, pos, v.clone().into())?;
             pos += 1;
         }
         Ok(())
     }
+
+    fn finalize_op<Obs: OpObserver>(
+        &mut self,
+        doc: &mut Automerge,
+        op_observer: &mut Obs,
```

Comment:
If we make this an `Option<&mut Obs>` then I think we can skip a lot of work in this method when there is no observer right?